src/app/cli/src/coda_main.ml

```diff
@@ -725,10 +730,13 @@ struct
           Transaction_snark_work.Checked.create_unsafe
             {Transaction_snark_work.fee; proofs= proof; prover} )
 
-    let load ~parent_log ~disk_location ~incoming_diffs =
+    let load ~parent_log ~disk_location ~incoming_diffs
+        ~frontier_broadcast_pipe =
       match%map Reader.load_bin_prot disk_location Pool.bin_reader_t with
-      | Ok pool -> of_pool_and_diffs pool ~parent_log ~incoming_diffs
-      | Error _e -> create ~parent_log ~incoming_diffs
+      | Ok pool ->
+          of_pool_and_diffs pool ~parent_log ~incoming_diffs
+            ~frontier_broadcast_pipe
```

Comment:
In this case, I think the deserialized snark_pool is referencing a stale frontier_broadcast_pipe and Network_pool.of_pool_and_diffs doesn't pass the new one in.